[PyCon DE Sprints] README: Update development setup instructions, general cleanup (#2519)
conda-forge/conda-smithy: README.md

> # Conda Smithy

Reviewer: ```suggestion
# conda-smithy
```

[![Ruff](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/astral-sh/ruff/main/assets/badge/v2.json)](https://github.com/astral-sh/ruff)
[![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/ambv/black)

## Installation

The easiest way to install conda-smithy is to use conda and conda-forge:

```
conda install -n root -c conda-forge conda-smithy
```

For development and contribution setup, see [Developing conda-smithy](#Developing-conda-smithy).

## Setup

You need a token from GitHub and potentially other services.
The commands which need this will tell you where to get these tokens and where to
place them.

> If you need help, please reach out to the [conda-forge Community](https://conda-forge.org/community/getting-in-touch/).

Reviewer: ```suggestion
If you need help, please [reach out](https://conda-forge.org/community/getting-in-touch/).
```

7. **Commit the changes:** ``cd foo-feedstock && git commit``, then push ``git push upstream master``.

> ## Running a build

Reviewer: This section probably belongs in `CONTRIBUTING.md` too.

Author: Are you sure? I thought this is part of the usage instructions: Setup -> Making a new feedstock -> _Running a build_

7. **Commit the changes:** ``cd foo-feedstock && git commit``, then push ``git push upstream master``.

## Running a build

When everything is configured, you can trigger a build with a push to the feedstock repo on GitHub.

## Developing conda-smithy

To install conda-smithy from source:

* Install `conda`, e.g., via [Miniforge](https://conda-forge.org/download/).

> * Clone this repository: `git clone https://github.com/conda-forge/conda-smithy` (or your fork).

Reviewer: Let's make sure folks do fork before.

```suggestion
* Fork and clone this repository: `git clone https://github.com/YOUR-USERNAME/conda-smithy.git`.
```

* Create a new conda environment with all requirements based on [environment.yml](environment.yml): `conda env create`.
* Activate the environment: `conda activate conda-smithy`.
  * Alternatively: Run `conda install -n base conda-spawn` and `conda spawn conda-smithy`
* Install conda-smithy: `pip install --no-deps --editable .`

To run all tests:

```sh
$ pytest
```

To run all code checks:

```sh
# staged changes
$ pre-commit run
# all files
$ pre-commit run --all-files
```

## Releasing conda-smithy

Before making a release, consult `@conda-forge/core` and wait some time for objections.

To release a new version of conda-smithy, you can use the
[rever](https://regro.github.io/rever-docs/index.html) release management tool.
Run `rever` in the root repo directory with the version number you want to release.
For example,

```sh
$ rever 0.1.2

> ```

Reviewer: This is now present in `CONTRIBUTING.md`, can you reconcile the differences and reflect them in that other file, removing this section? Thanks!

Author: Done in ddbd2acf9b4f1bc8ea8986ed7c79504fb5f1e7b2. Sorry.
